csharp-advanced-topics/ActionInCSharp/TestScenario/UnitTestAction.cs

```diff
@@ -0,0 +1,43 @@
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+
+namespace TestScenario
+{
+    [TestClass]
+    public class UnitTestAction
+    {
+        private static int resultNumber = 0;
+        private static string resultMessage = "";
+        [TestMethod]
+        public void When_PassingSingleValue()
+        {
```

Comment:
Test names are almost what they should be. When part should be more descriptive, and you need a then part. Please check the guidelines for proper test naming.